### PR TITLE
libsForQt5.applet-window-appmenu: init at unstable-2022-06-27

### PR DESCRIPTION
**Pull request by greydot**
## Description of changes
Add applet-window-appmenu package. This library adds global menu similar to the stock KDE one, but with more options. From latte-dock author.

The last release doesn't build work with KDE 5.27, but this code does.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by stigtsp — APPROVED** [review]
LGTM, but not using Plasma so would be good if someone could double check

**Review by SuperSandro2000 — APPROVED** [review]
codewise LGTM, not sure how to test this

**Comment by greydot** [comment]
> codewise LGTM, not sure how to test this

When using KDE Plasma, add this to environment.systemPackages, install, and add as a widget to a panel. The widget will be called "Window AppMenu". It'll look something like this.

https://ibb.co/JFfFtGz

File: pkgs/development/libraries/applet-window-appmenu/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+, plasma-workspace
+, libSM
+, qtx11extras
+, kwindowsystem
+, libdbusmenu
+}:
+
+mkDerivation rec {
+  pname = "applet-window-appmenu";
+  version = "unstable-27-06-2022";
```
Comment (wegank) [inline pkgs/development/libraries/applet-window-appmenu/default.nix:19]:
```suggestion
  version = "unstable-2022-06-27";
```

Comment (greydot) [reply pkgs/development/libraries/applet-window-appmenu/default.nix]: Fixed.

File: pkgs/development/libraries/applet-window-appmenu/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    libSM
+    qtx11extras
+    libdbusmenu
+  ];
+
+  meta = with lib; {
+    description = "Plasma 5 applet in order to show window menu in your panels";
+    homepage = "https://github.com/psifidotos/applet-window-appmenu";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ greydot ];
```
Comment (wegank) [inline pkgs/development/libraries/applet-window-appmenu/default.nix:49]:
```suggestion
    maintainers = with maintainers; [ greydot ];
    platforms = platforms.linux;
```

File: pkgs/development/libraries/applet-window-appmenu/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+, kdecoration
+, plasma-framework
+, plasma-workspace
+, libSM
+, qtx11extras
+, kwindowsystem
+, libdbusmenu
+}:
+
+mkDerivation rec {
```
Comment (wegank) [inline pkgs/development/libraries/applet-window-appmenu/default.nix:17]: Could you try using `stdenv.mkDerivation` here? #180841

Comment (greydot) [reply pkgs/development/libraries/applet-window-appmenu/default.nix:17]: It doesn't build that way.
```
qtPreHook
Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.
error: builder for '/nix/store/gk00f0zyxbr9rc3hnrmfk2da5isaby6r-applet-window-appmenu-unstable-2022-06-27.drv' failed with exit code 1;
       last 2 log lines:
       > qtPreHook
       > Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.
       For full logs, run 'nix log /nix/store/gk00f0zyxbr9rc3hnrmfk2da5isaby6r-applet-window-appmenu-unstable-2022-06-27.drv'.
```

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 259478` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libsForQt5.applet-window-appmenu (plasma5Packages.applet-window-appmenu)</li>
  </ul>
</details>

**Pull request merged**